src/test/test.h

```diff
@@ -93,3 +94,21 @@ int test_main(struct test *start, struct test *stop, void (*reset_func)(void), i
             exit(-1);                                                   \
         }                                                               \
     } while (0)
+
+#define ASSERT_STR_EQ(str_a, str_b)                                     \
+    do {                                                                \
+        if (strcmp((char *)str_a, (char *)str_b) != 0) {                \
```

Comment:
should we cast here? Giving it an incorrect pointer would crash the test while no compiler errors were emitted. I'd also make these macros smart and check for NULL, printing proper error messages and not crashing :-)